src/virtualenv/seed/embed/base_embed.py

```diff
@@ -98,6 +109,15 @@ def add_parser_arguments(cls, parser, interpreter, app_data):
             help="disable the periodic (once every 14 days) update of the embedded wheels",
             default=not PERIODIC_UPDATE_ON_BY_DEFAULT,
         )
+        parser.add_argument(
```

Comment:
IMHO this should be a separate plugin. This needs a lot more thought, on e.g. how this interacts with periodic update feature, transitive dependencies and where do we get these custom wheels you're injecting here. Perhaps you're after a pip functionality here, not virtualenv one. In general would be nice to open an issue to discuss such proposal before you lead on with a solution.

Reply:
> IMHO this should be a separate plugin. This needs a lot more thought, on e.g. how this interacts with periodic update feature, transitive dependencies and where do we get these custom wheels you're injecting here. Perhaps you're after a pip functionality here, not virtualenv one. In general would be nice to open an issue to discuss such proposal before you lead on with a solution.

Should the discussion be here? I find it's often useful to propose a problem and solution together.

The wheels are supplied externally, hence my `VIRTUALENV_EXTRA_SEARCH_DIR`.

It shouldn't interact with periodic update; AFAICT the intent of that is only for the embedded wheels (maybe I should ensure that code is explicitly restricted to those).

Based on #1923 it seems like virtualenv won't support traversing dependencies for now. Until it does, you need to specify your whole dependency set in `--seed-package`, but this seems fine for now.

Updating pip would solve my problem as well; I thought that `--seed-package` was a smaller ask than `pip` vendoring more code. I thought they would want to be as lightweight as possible. If you think I'm more likely to succeed with pip, I can try there instead.

Reply:
> Based on #1923 it seems like virtualenv won't support traversing dependencies for now. 

This is not true, just waiting for implementation. I'd consider this a blocking functionality request for this PR.

>  Until it does, you need to specify your whole dependency set in --seed-package, but this seems fine for now.

I disagree.

> Updating pip would solve my problem as well; I thought that `--seed-package` was a smaller ask than `pip` vendoring more code.

Wishful thinking. 😊 

If you want to go ahead with the current works for your half-baked solution I recommend using a plugin, but for acceptance in the core, we'd need at the very least traversing and handling dependencies too.

Reply:
> If you want to go ahead with the current works for your half-baked solution I recommend using a plugin, but for acceptance in the core, we'd need at the very least traversing and handling dependencies too.

Fair enough! thanks for your time.